### PR TITLE
Openshift 4.x

### PR DESCRIPTION
**Pull request by singularo**
Catch up to develop and add the default .gitlab-ci.yml

### CODE REVIEW

File: scaffold/optional/.gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    when: on_failure
+  except:
+    - master
+    - develop
+  script:
+    # Set the hostname for selenium tests.
+    - export SIMPLETEST_BASE_URL="http://$(ip a | grep global | awk '{ print $2 }' | cut -d'/' -f1):8080"
+    - export DTT_BASE_URL="http://$(ip a | grep global | awk '{ print $2 }' | cut -d'/' -f1):8080"
+    # Use https clone URLs.
+    - git config --global url."https://${COMPOSER_USER}:${COMPOSER_TOKEN}@gitlab.adelaide.edu.au/".insteadOf "git@gitlab.adelaide.edu.au:"
```
Comment (singularo) [inline scaffold/optional/.gitlab-ci.yml:58]: I don't like that this is here, but it wont actually change anything if we get external people using this, as there will be no matches, so its safe to have here, just a little polluting.

File: scaffold/optional/.gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    name:  selenium/standalone-chrome:99.0
+
+variables:
+  # build directory.
+  HOME: "$CI_BUILDS_DIR"
+  # "db" env vars.
+  DATABASE_HOST: db
+  MYSQL_DATABASE: drupal
+  MYSQL_USER: user
+  MYSQL_PASSWORD: password
```
Comment (pingers) [inline scaffold/optional/.gitlab-ci.yml:22]: These passwords don't belong here. Can't we inject via GitLab?

Comment (pingers) [reply scaffold/optional/.gitlab-ci.yml:22]: E.g. https://gitlab.adelaide.edu.au/web-team/ua-wcms-foundation/-/settings/ci_cd#js-cicd-variables-settings

Comment (pingers) [reply scaffold/optional/.gitlab-ci.yml:22]: Oh, we're using the local db anyway... Fair enough :smile:

File: scaffold/optional/.gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  SHEPHERD_TOKEN: super-secret-token
+  SITE_TITLE: Shepherd Drupal Project
+  SITE_MAIL: site@example.com
+  SITE_ADMIN_EMAIL: admin@example.com
+  SITE_ADMIN_USERNAME: admin
+  SITE_ADMIN_PASSWORD: password
+
+build:
+  tags:
+    - openshift
```
Comment (pingers) [inline scaffold/optional/.gitlab-ci.yml:44]: I think this should still be drupal?

Comment (pingers) [reply scaffold/optional/.gitlab-ci.yml:44]: Or is it very generic openshift runner now?

File: scaffold/optional/.gitlab-ci.yml (hunk 1 above)
Comment (pingers) [reply scaffold/optional/.gitlab-ci.yml:58]: git requires ssh key, http just needs a token, which is easier to manage.

**Comment by pingers** [comment]
Closing in favour of https://github.com/universityofadelaide/shepherd-drupal-scaffold/pull/124

**Pull request closed without merging**